tests/filehandler/filehandler_test.go

```diff
@@ -0,0 +1,55 @@
+package files_test
+
+import (
+	"gz-beat-shipper/pkg/filehandler"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Check GNU zip decompress func works
+func TestGzipFileReader(t *testing.T) {
+	gzTestFileName := "test.txt.gz"
+	gzipReader := filehandler.GzipFileReader(gzTestFileName)
+	text, _ := ioutil.ReadAll(gzipReader)
+	assert := assert.New(t)
+	assert.Equal("This is the content of the compressed file\n", string(text))
+}
+
+// Check gzipReader copy the content to the destination file
+// It should create it and remove it
+func TestCopyGzipContentToFile(t *testing.T) {
+	gzTestFileName := "test.txt.gz"
+	destinationFile := "test.txt"
+	gzipReader := filehandler.GzipFileReader(gzTestFileName)
+	filehandler.CopyGzipContentToFile(gzipReader, destinationFile)
+
+	assert := assert.New(t)
+	assert.True(existFile(destinationFile))
+
+	_ = os.Remove(destinationFile)
+}
+
+// Test if remove the last extension in a right way
+func TestGetFileNameWithoutLastExtension(t *testing.T) {
+	gzFileName := "test.txt.gz"
+	txtFileName := filehandler.GetFileNameWithoutLastExtension(gzFileName)
+	assert := assert.New(t)
+	assert.Equal("test.txt", txtFileName)
+	fileName := filehandler.GetFileNameWithoutLastExtension(txtFileName)
+	assert.Equal("test", fileName)
+}
+
+// There's no a direct way to check if file exists in Go
+// So we should stat it with a call to the system
+func existFile(fileName string) bool {
```

Comment:
You are depending on the filesystem for these tests, you can consider them to be integration tests then. Don't need to move them like I mentioned before. 

If you would like them to be unit tests, have existFile be a mock that answers as you please per test.